tests/filecheck/frontend/pypdl/pdl.py

```diff
@@ -0,0 +1,73 @@
+# RUN: python %s | filecheck %s
+
+"""Tests for PDL rewriting Python DSL.
+
+These tests exercise the API providing an evaluatable Python DSL for expressing
+the semantics of rewrites, which can then be lowered to the PDL dialect using
+the PyAST frontend.
+
+The following code snippets show early suggestions for the structure of this API:
+
+- <https://github.com/xdslproject/xdsl/pull/1137>
+- <https://github.com/opencompl/xdsl-smt/blob/main/xdsl_smt/semantics/llvm_semantics.py#L225>
+
+This example exercises the simple case of erasing a matched operation:
+
+```mlir
+pdl.pattern : benefit(2) {
+  %0 = pdl.type
+  %1 = pdl.attribute = 0 : i32
+  %2 = pdl.operation "arith.constant" {"value" = %1} -> (%0 : !pdl.type)
+  pdl.rewrite %2 {
+    pdl.erase %2
+  }
+}
+```
+"""
+
+from xdsl.dialects import arith, builtin, pdl
+from xdsl.frontend.pypdl.context import PyPDLContext
+from xdsl.ir import Operation
+from xdsl.rewriter import Rewriter
+
+
+def erase_op(operation: Operation) -> None:
+    """Shim to avoid `Expr` AST node required for methods."""
+    return Rewriter.erase_op(operation)
+
+
+ctx = PyPDLContext()
+ctx.register_type(arith.ConstantOp, pdl.OperationType())
+ctx.register_function(erase_op, pdl.EraseOp)
```

Comment:
shouldn't these always be registered for things to work? I would recommend putting this into the init of PyPDLContext